ProcessLib/ThermoMechanics/ThermoMechanicsFEM.h

```diff
@@ -0,0 +1,545 @@
+                              std::vector<double> const& local_xdot,
+                              const double /*dxdot_dx*/, const double /*dx_dx*/,
+                              std::vector<double>& /*local_M_data*/,
+                              std::vector<double>& /*local_K_data*/,
+                              std::vector<double>& local_rhs_data,
+                              std::vector<double>& local_Jac_data) override
+    {
+        auto const local_matrix_size = local_x.size();
+        assert(local_matrix_size == temperature_size + displacement_size);
+
+        auto T = Eigen::Map<typename ShapeMatricesType::template VectorType<
+            temperature_size> const>(local_x.data() + temperature_index,
+                                     temperature_size);
+
+        auto u = Eigen::Map<typename ShapeMatricesType::template VectorType<
+            displacement_size> const>(local_x.data() + displacement_index,
+                                      displacement_size);
+
+        auto T_dot = Eigen::Map<typename ShapeMatricesType::template VectorType<
+            temperature_size> const>(local_xdot.data() + temperature_index,
+                                     temperature_size);
+
+        auto local_Jac = MathLib::createZeroedMatrix<JacobianMatrix>(
+            local_Jac_data, local_matrix_size, local_matrix_size);
+
+        auto local_rhs = MathLib::createZeroedVector<RhsVector>(
+            local_rhs_data, local_matrix_size);
+
+        typename ShapeMatricesType::template MatrixType<displacement_size,
+                                                        temperature_size> KuT;
+        KuT.setZero(displacement_size, temperature_size);
+
+        typename ShapeMatricesType::NodalMatrixType KTT;
+        KTT.setZero(temperature_size, temperature_size);
+
+        typename ShapeMatricesType::NodalMatrixType DTT;
+        DTT.setZero(temperature_size, temperature_size);
+
+        double const& dt = _process_data.dt;
+
+        SpatialPosition x_position;
+        x_position.setElementID(_element.getID());
+
+        unsigned const n_integration_points =
+            _integration_method.getNumberOfPoints();
+        for (unsigned ip = 0; ip < n_integration_points; ip++)
+        {
+            x_position.setIntegrationPoint(ip);
+            auto const& w = _ip_data[ip].integration_weight;
+
+            auto const& dNdx = _ip_data[ip].dNdx;
+            auto const& N = _ip_data[ip].N;
+
+            auto const& B = _ip_data[ip].b_matrices;
+            auto const& sigma = _ip_data[ip].sigma;
+
+            auto const& C = _ip_data[ip].C;
+
+            double const delta_T = N.dot(T) - _process_data.reference_temperature;
+            // calculate thermally induced strain
+            auto const alpha =
+                _process_data.linear_thermal_expansion_coefficient(
+                    t, x_position)[0];
+            double const thermal_strain = alpha * delta_T;
```

Comment:
call this (here and where you pass it) "linear_thermal_strain". Again, a comment "assuming isotropic alpha"